This patch is based on the upstream commit described below, adapted for
use in the Debian package and to further bump the nom dependency to
version 7. 

commit 4d8d633013b4f87d0790568731d09a3e97cd5163
Author: kpcyrd <git@rxv.cc>
Date:   Thu May 6 17:49:23 2021 +0200

    Port to nom 6

Index: config/src/path/parser.rs
===================================================================
--- config.orig/src/path/parser.rs
+++ config/src/path/parser.rs
@@ -33,7 +33,7 @@ fn ident(i: &str) -> IResult<&str, Expre
     map(raw_ident, Expression::Identifier)(i)
 }
 
-fn postfix<'a>(expr: Expression) -> impl Fn(&'a str) -> IResult<&'a str, Expression> {
+fn postfix<'a>(expr: Expression) -> impl FnMut(&'a str) -> IResult<&'a str, Expression> {
     let e2 = expr.clone();
     let child = map(preceded(tag("."), raw_ident), move |id| {
         Expression::Child(Box::new(expr.clone()), id)
@@ -71,10 +71,10 @@ pub fn from_str(input: &str) -> Result<E
     }
 }
 
-pub fn to_error_kind(e: Err<(&str, ErrorKind)>) -> ErrorKind {
+pub fn to_error_kind(e: Err<nom::error::Error<&str>>) -> ErrorKind {
     match e {
         Err::Incomplete(_) => ErrorKind::Complete,
-        Err::Failure((_, e)) | Err::Error((_, e)) => e,
+        Err::Failure(e) | Err::Error(e) => e.code,
     }
 }
 
Index: config/Cargo.toml
===================================================================
--- config.orig/Cargo.toml
+++ config/Cargo.toml
@@ -27,3 +27,3 @@ repository = "https://github.com/mehcode
 [dependencies.nom]
-version = "5.0.0"
+version = "7.0.0"
 
